File: 242.valid-anagram.cs
/*
 * @lc app=leetcode id=242 lang=csharp
 *
 * [242] Valid Anagram
 */

// @lc code=start
public class Solution
{
    public bool IsAnagram(string s, string t)
    {
        if (s.Length != t.Length)
            return false;
        Dictionary<char, int> pair = new Dictionary<char, int>();
        foreach (char c in s)
        {
            if (pair.ContainsKey(c))
                pair[c]++;
            else
                pair[c] = 1;
        }
        foreach (char c in t)
        {
            if (pair.ContainsKey(c))
                pair[c]--;
            else
                return false;
        }
        foreach (var k in pair)
        {
            if (k.Value != 0)
                return false;
        }
        return true;

    }
}
// @lc code=end

File: 242.valid-anagram.py
#
# @lc app=leetcode id=242 lang=python3
#
# [242] Valid Anagram
#

# @lc code=start
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        
        if len(s) != len(t):
            return False
        
        pair = {}

        for char in s:
            pair[char] = pair.get(char, 0) + 1

        for char in t:
            if pair.get(char, 0) == 0:
                return False
            else:
                pair[char] -= 1
        
        return True
                
# @lc code=end

